### PR TITLE
Initial support for <defs> and <use> tags in SVG imports

### PR DESCRIPTION
**Pull request by alexjlockwood**
This pull request branches off of pull request #58, so please review that first.

Fun fact: this is the first time I've ever actually gotten an Adobe Illustrator generated SVG imported correctly using ANY converter software... so yay!

### CODE REVIEW

File: app/scripts/svgloader.js [hunk 1]
```diff
@@ -28,15 +28,51 @@ export const SvgLoader = {
 
     let usedIds = {};
 
+    let defsChildMap = new Map();
+    let extractDefsMap_ = (node) => {
+      if (!node || !node.childNodes.length) {
+        return;
+      }
+      for (let i = 0; i < node.childNodes.length; i++) {
```
Comment (alexjlockwood) [inline app/scripts/svgloader.js]: TODO: replace this with something more functional?

File: app/scripts/colorutil.js [hunk 2]
```diff
@@ -59,17 +59,14 @@ export const ColorUtil = {
     return str;
   },
 
-  svgToAndroidColor(color, opacity) {
```
Comment (romannurik) [inline app/scripts/colorutil.js]: this is from a different pull request, right? mind merging?

File: app/scripts/svgloader.js [hunk 3]
```diff
@@ -125,9 +161,9 @@ export const SvgLoader = {
         return Object.assign(layerData, {
           id: makeFinalNodeId_('path'),
           pathData: path,
-          fillColor: ColorUtil.svgToAndroidColor(context.fillColor, context.fillOpacity),
+          fillColor: ('fillColor' in context) ? ColorUtil.svgToAndroidColor(context.fillColor) : null,
```
Comment (romannurik) [inline app/scripts/svgloader.js:166]: from a different PR

File: app/scripts/svgpathdata.js [hunk 4]
```diff
@@ -79,7 +79,13 @@ export class SvgPathData {
   transform(pointTransformer) {
     this.commands_.forEach(({command, args}) => {
       if (command == '__arc__') {
-        // TODO: transform arcs
+        // TODO: apply the transformation to the middle indicies as well.
```
Comment (romannurik) [inline app/scripts/svgpathdata.js]: split this out to a different PR?

Comment (alexjlockwood) [reply app/scripts/svgpathdata.js]: Sure thing.

BTW, I'm not missing anything here right? As in even though it still doesn't cover all cases... this additional code should still improve things, right?

File: app/scripts/colorutil.js (hunk 2 above)
Comment (alexjlockwood) [reply app/scripts/colorutil.js]: Wow, I can't believe I successfully merged on my first try w/o screwing everything up. \o/

File: app/scripts/svgpathdata.js (hunk 4 above)
Comment (alexjlockwood) [reply app/scripts/svgpathdata.js]: Reverted. Will work on improving separately.

**Comment by alexjlockwood** [comment]
BTW, if you ever pick this up again, this CL doesn't handle the case where some attributes are specified on the `<use>` tag.
